### PR TITLE
Handle ErrorTypeSymbol as a ContainingSymbol of TypeParameterSymbol in CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitTypeParameter.

### LINKED ISSUE #8523: InvalidOperationException is thrown in CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitTypeParameter
Ported from DevDiv 187859

>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitTypeParameter(Microsoft.CodeAnalysis.CSharp.Symbols.TypeParameterSymbol symbol, System.Text.StringBuilder builder) Line 165	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.TypeParameterSymbol.Accept<System.__Canon, System.__Canon>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon> visitor, System.__Canon argument) Line 184	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon>.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol, System.__Canon argument) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol symbol, System.Text.StringBuilder builder) Line 207	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitErrorType(Microsoft.CodeAnalysis.CSharp.Symbols.ErrorTypeSymbol symbol, System.Text.StringBuilder builder) Line 255	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ErrorTypeSymbol.Accept<System.__Canon, System.__Canon>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon> visitor, System.__Canon argument) Line 349	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon>.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol, System.__Canon argument) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitParameter(Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol symbol, System.Text.StringBuilder builder) Line 245	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol.Accept<System.__Canon, System.__Canon>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon> visitor, System.__Canon argument) Line 240	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon>.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol, System.__Canon argument) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitParameters(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol> parameters, bool isVararg, System.Text.StringBuilder builder) Line 83	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.PartVisitor.VisitMethod(Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol symbol, System.Text.StringBuilder builder) Line 111	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol.Accept<System.__Canon, System.__Canon>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon> visitor, System.__Canon argument) Line 621	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon>.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol, System.__Canon argument) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentIDVisitor.VisitMethod(Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol symbol, System.Text.StringBuilder builder) Line 38	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol.Accept<System.__Canon, System.__Canon>(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon> visitor, System.__Canon argument) Line 621	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor<System.__Canon, System.__Canon>.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol, System.__Canon argument) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.GetDocumentationCommentId() Line 725	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.GetDocumentationCommentId(Microsoft.CodeAnalysis.CSharp.Syntax.CrefSyntax crefSyntax, Microsoft.CodeAnalysis.CSharp.Binder binder, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 961	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DocumentationCommentWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 91	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor.VisitXmlCrefAttribute(Microsoft.CodeAnalysis.CSharp.Syntax.XmlCrefAttributeSyntax node) Line 69185	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.XmlCrefAttributeSyntax.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor visitor) Line 64873	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.Visit(Microsoft.CodeAnalysis.SyntaxNode node) Line 35	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 54	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DocumentationCommentWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 159	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor.VisitXmlEmptyElement(Microsoft.CodeAnalysis.CSharp.Syntax.XmlEmptyElementSyntax node) Line 69161	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.XmlEmptyElementSyntax.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor visitor) Line 64490	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.Visit(Microsoft.CodeAnalysis.SyntaxNode node) Line 35	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 54	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DocumentationCommentWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 159	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor.VisitXmlElement(Microsoft.CodeAnalysis.CSharp.Syntax.XmlElementSyntax node) Line 69143	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.XmlElementSyntax.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor visitor) Line 64196	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.Visit(Microsoft.CodeAnalysis.SyntaxNode node) Line 35	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 54	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DocumentationCommentWalker.DefaultVisit(Microsoft.CodeAnalysis.SyntaxNode node) Line 159	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor.VisitDocumentationCommentTrivia(Microsoft.CodeAnalysis.CSharp.Syntax.DocumentationCommentTriviaSyntax node) Line 69083	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Syntax.DocumentationCommentTriviaSyntax.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxVisitor visitor) Line 63218	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSyntaxWalker.Visit(Microsoft.CodeAnalysis.SyntaxNode node) Line 35	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DocumentationCommentWalker.GetSubstitutedText(Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Microsoft.CodeAnalysis.CSharp.Symbol symbol, Microsoft.CodeAnalysis.CSharp.Syntax.DocumentationCommentTriviaSyntax trivia, Microsoft.CodeAnalysis.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode> includeElementNodes, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol> documentedParameters, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbols.TypeParameterSymbol> documentedTypeParameters) Line 75	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.TryProcessDocumentationCommentTriviaNodes(Microsoft.CodeAnalysis.CSharp.Symbol symbol, bool isPartialMethodDefinitionPart, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.Syntax.DocumentationCommentTriviaSyntax> docCommentNodes, bool reportParameterOrTypeParameterDiagnostics, out string withUnprocessedIncludes, out bool haveParseError, out System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbols.TypeParameterSymbol> documentedTypeParameters, out System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol> documentedParameters, out System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CSharp.CSharpSyntaxNode> includeElementNodes) Line 436	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.DefaultVisit(Microsoft.CodeAnalysis.CSharp.Symbol symbol) Line 284	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor.VisitMethod(Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol symbol) Line 64	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MethodSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 627	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol symbol) Line 214	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 576	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.VisitNamespace(Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol symbol) Line 183	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceSymbol.Accept(Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor visitor) Line 136	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpSymbolVisitor.Visit(Microsoft.CodeAnalysis.CSharp.Symbol symbol) Line 15	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.DocumentationCommentCompiler.WriteDocumentationCommentXml(Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, string assemblyName, System.IO.Stream xmlDocStream, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, System.Threading.CancellationToken cancellationToken, Microsoft.CodeAnalysis.SyntaxTree filterTree, Microsoft.CodeAnalysis.Text.TextSpan? filterSpanWithinTree) Line 95	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnosticsForMethodBodiesInTree(Microsoft.CodeAnalysis.SyntaxTree tree, Microsoft.CodeAnalysis.Text.TextSpan? span, System.Threading.CancellationToken cancellationToken) Line 2056	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetDiagnosticsForSyntaxTree(Microsoft.CodeAnalysis.CompilationStage stage, Microsoft.CodeAnalysis.SyntaxTree syntaxTree, Microsoft.CodeAnalysis.Text.TextSpan? filterSpanWithinTree, bool includeEarlierStages, System.Threading.CancellationToken cancellationToken) Line 2202	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDiagnostics(Microsoft.CodeAnalysis.Text.TextSpan? span, System.Threading.CancellationToken cancellationToken) Line 126	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.CompilationData.GetOrCreateCachedSemanticModel(Microsoft.CodeAnalysis.SyntaxTree tree, Microsoft.CodeAnalysis.Compilation compilation, System.Threading.CancellationToken cancellationToken) Line 55	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSemanticDiagnosticsCoreAsync.AnonymousMethod__0() Line 491	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.ComputeAnalyzerDiagnosticsAsync(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope analysisScope, System.Action generateCompilationEventsOpt, System.Func<Microsoft.CodeAnalysis.Diagnostics.AsyncQueue<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>> getEventQueue, int newTaskToken, System.Threading.CancellationToken cancellationToken) Line 526	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSemanticDiagnosticsCoreAsync(Microsoft.CodeAnalysis.SemanticModel model, Microsoft.CodeAnalysis.Text.TextSpan? filterSpan, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer> analyzers, System.Threading.CancellationToken cancellationToken) Line 499	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.GetAnalyzerSemanticDiagnosticsAsync(Microsoft.CodeAnalysis.SemanticModel model, Microsoft.CodeAnalysis.Text.TextSpan? filterSpan, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer> analyzers, System.Threading.CancellationToken cancellationToken) Line 478	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticAnalyzerDriver.GetSemanticDiagnosticsAsync(Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer) Line 230	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.GetSemanticDiagnosticsAsync(Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticAnalyzerDriver userDiagnosticDriver, Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer analyzer) Line 881	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.AnalyzerExecutor.GetDocumentAnalysisDataAsync(Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticAnalyzerDriver analyzerDriver, Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.StateSet stateSet, Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.VersionArgument versions) Line 70	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.AnalyzeDocumentAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.VersionArgument versions, System.Collections.Immutable.ImmutableHashSet<string> diagnosticIds, bool skipClosedFileChecks, System.Threading.CancellationToken cancellationToken) Line 279	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.AnalyzeDocumentAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.SyntaxNode bodyOpt, System.Collections.Immutable.ImmutableHashSet<string> diagnosticIds, bool skipClosedFileChecks, System.Threading.CancellationToken cancellationToken) Line 195	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.AnalyzeDocumentAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.SyntaxNode bodyOpt, System.Threading.CancellationToken cancellationToken) Line 178	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzerService.IncrementalAnalyzerDelegatee.AnalyzeDocumentAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.SyntaxNode bodyOpt, System.Threading.CancellationToken cancellationToken) Line 86	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.ProcessDocumentAnalyzersAsync.AnonymousMethod__28_1(Microsoft.CodeAnalysis.SolutionCrawler.IIncrementalAnalyzer a, Microsoft.CodeAnalysis.Document d, System.Threading.CancellationToken c) Line 184	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.RunAnalyzersAsync.AnonymousMethod__0(System.__Canon v, System.Threading.CancellationToken c) Line 206	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.GetOrDefaultAsync<System.__Canon, System.__Canon>(System.__Canon value, System.Func<System.__Canon, System.Threading.CancellationToken, System.Threading.Tasks.Task<System.__Canon>> funcAsync, System.Threading.CancellationToken cancellationToken)	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.RunAnalyzersAsync<System.__Canon>(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.SolutionCrawler.IIncrementalAnalyzer> analyzers, System.__Canon value, System.Func<Microsoft.CodeAnalysis.SolutionCrawler.IIncrementalAnalyzer, System.__Canon, System.Threading.CancellationToken, System.Threading.Tasks.Task> runnerAsync, System.Threading.CancellationToken cancellationToken) Line 204	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.IncrementalAnalyzerProcessor.ProcessDocumentAnalyzersAsync(Microsoft.CodeAnalysis.Document document, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.SolutionCrawler.IIncrementalAnalyzer> analyzers, Microsoft.CodeAnalysis.SolutionCrawler.SolutionCrawlerRegistrationService.WorkCoordinator.WorkItem workItem, System.Threading.CancellationToken cancellationToken) Line 184	C#

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #8523.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/DocumentationComments/DocumentationCommentIDVisitor.PartVisitor.cs [hunk 1]
```diff
@@ -146,8 +146,14 @@ public override object VisitTypeParameter(TypeParameterSymbol symbol, StringBuil
...
                 // Is this a type parameter on a type?
                 Symbol containingSymbol = symbol.ContainingSymbol;
-                if (containingSymbol.Kind == SymbolKind.NamedType)
+                if (containingSymbol.Kind == SymbolKind.Method)
                 {
+                    builder.Append("``");
+                }
+                else 
+                {
+                    Debug.Assert(containingSymbol is NamedTypeSymbol);
```
Comment (cston) [inline src/Compilers/CSharp/Portable/DocumentationComments/DocumentationCommentIDVisitor.PartVisitor.cs]: Is there a test for the ErrorTypeSymbol case?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/DocumentationComments/DocumentationCommentIDVisitor.PartVisitor.cs]: No there is not and I couldn't come up with one, none of the Watson dumps include heap. The code in the block should work fine for ErrorTypeSymbols, in fact it doesn't even depend on the type or kind of this symbol.

**Comment by cston** [comment]
LGTM

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by jaredpar** [comment]
:+1:

**Pull request merged**